sqlmesh/core/context_diff.py

```diff
@@ -406,3 +394,33 @@ def text_diff(self, name: str) -> str:
         except SQLMeshError as e:
             logger.warning("Failed to diff model '%s': %s", name, str(e))
             return ""
+
+
+def _build_requirements(
```

Comment:
why was this moved out?

Reply:
Cause context_diff should only care about the final set of requirements. Because of lazy computation I had to pass both `provided_requirements` and `excluded_requirements` into the instance itself just so that the actual value is computed later. This presented additional complications when it came to creating the `no_diff` instance, since I had to pass the `excluded_requirements` even though requirements were sourced from the target environment.